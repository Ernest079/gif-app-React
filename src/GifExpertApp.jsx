import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {
  const [categories, setcategories] = useState([
    'Haikyuu'
  ]);

  const onAddCategory = (newCategory) => {
    // console.log(newCategory);
    if(categories.includes(newCategory)){
      return;
    }
    setcategories([newCategory, ...categories]);
  }

  return (
    <>
      <h1>GifExpertApp</h1>
      {/* Ya */}
      <AddCategory 
        onNewCategory = {(value) => onAddCategory(value)}
      />
      {categories.map((category) => (
        // Ya
        <GifGrid 
          key = {category} 
          category={category}
        />
      ))}
    </>
  );
}
