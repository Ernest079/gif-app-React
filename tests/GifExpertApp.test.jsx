import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe('Pruebas en GifExpertAPP', () => {
  test('Debe hacer match con el snapshot', () => {
    const {container} = render(<GifExpertApp/>);
    // screen.debug();
    expect(container).toMatchSnapshot();
  });

  test('Debe de agregar la categoria', () => {
    render(<GifExpertApp/>);
    const value = 'Naruto';

    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form', {name: "gif-form"});

    fireEvent.input(input, {target:{value: value}});
    // screen.debug(); 
    fireEvent.submit(form);
    // screen.debug(); 

    expect(screen.getByText(value)).toBeTruthy();
  });
});