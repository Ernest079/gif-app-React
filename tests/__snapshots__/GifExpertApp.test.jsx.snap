// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertAPP Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="gif-form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h2>
    Haikyuu
  </h2>
  <h4>
    Cargando...
  </h4>
  <div
    class="card-grid"
  />
</div>
`;
